src/pages/guides/webpack-configuration.md

```diff
@@ -0,0 +1,194 @@
+  
+### Async function that returns an array of objects
+  ```
+  module.exports = async (env) => ([
+    {
+      mode: 'development' 
+    },
+    {
+      mode: 'production'
+    }
+  ])
+  ```
+### Array of objects
+  ```
+  module.exports = [
+    {
+      mode: 'development'
+    },
+    {
+      mode: 'production'
+    }
+  ]
+  ```
+  
+### Array of functions that return objects
+  ```
+  module.exports = [
+    (env) => ({
+      mode: 'development'
+    }),
+    (env) => ({
+      mode: 'production'
+    })
+  ]
+  ```
+  
+### Array of async functions that return objects
+  ```
+  module.exports = [
+    async (env) => ({
+      mode: 'development'
+    }),
+    async (env) => ({
+      mode: 'production'
+    })
+  ]
+  ```
+ 
+## Configuration
+
+### Overview
+
+Assuming `config` contains your Webpack configuration: 
+
+```
+{
+  mode: config.mode || 'production', 
+  optimization: {
+    minimize: config.optimization.minimize || false
+  },
+  output: { 
+    libraryTarget: config.output.libraryTarget || 'commonjs2',
+    path: './dist/${actionPath}/${actionName}-temp/' // Cannot change 
+    fileName: `${output.path}/index.js` // Cannot change 
+  },
+  target: 'node' // Cannot change,
+  entry: [
+    <path to the action>, 
+    ...config.entry
+  ],
+  resolve: { 
+    extensions: [
+      '.js', 
+      '.json', 
+      ...config.extensions
+    ],
+    mainFields: [ 
+      'main',
+      ...config.mainFields
+    ]
+  },
+  plugins: [
+    new DefaultPlugin({
+      WEBPACK_ACTION_BUILD: 'true', 
+      `process.env.AIO_CLI_ENV`: ${cliEnv}
+    }),
+    ...config.plugins
+  ]
+  
+  ... all other Webpack configuration options are valid
+
+}
+```
+
+### Base Options 
+
+App Builder starts off with the following Webpack configuration options, any additional values will be added on for these fields: 
+
+* `entry` - Will always contain the path to the action, additional paths identified in your configuration file will be added on.
+* `resolve.extensions` - Will always contain `.js` and `.json`, additional extensions identified in your configuration file will be added on.
+* `resolve.mainFields` - Will always contain `main`, additional main fields identified in your configuration file will be added on.
+* `plugins` - Will always contain a plugin that injects the following, additional plugins identified in your configuratoin file will be added on.
+   
+   ```
+   {
+      WEBPACK_ACTION_BUILD: 'true', 
+      `process.env.AIO_CLI_ENV`: ${cliEnv}
+   }
+   ```
+   
+### Defaults 
+
+App Builder defaults the following Webpack configuration options if they are not present in your configuration file: 
+
+* `output.libraryTarget` - Will default to `commonjs2`.
+* `mode` - Will default to `production`. 
+* `optimization.minimize` - Will default to `false`.
+
+### Immutable Options 
+
+App Builder does not allow changing the following Webpack configuration options. These fields will be ignored if present in your configuration file: 
```

Comment:
`App Builder does not allow changing of the following Webpack configuration options.`